atom/browser/api/atom_api_web_contents.cc

```diff
@@ -345,15 +345,25 @@ void WebContents::DidGetResourceResponseStart(
   v8::Isolate* isolate = v8::Isolate::GetCurrent();
   v8::Locker locker(isolate);
   v8::HandleScope handle_scope(isolate);
-  mate::Dictionary response_headers(isolate, v8::Object::New(isolate));
+  base::DictionaryValue response_headers;
 
   net::HttpResponseHeaders* headers = details.headers.get();
   void* iter = nullptr;
   std::string key;
   std::string value;
-  while (headers && headers->EnumerateHeaderLines(&iter, &key, &value))
-    response_headers.Set(base::StringToLowerASCII(key),
-                         base::StringToLowerASCII(value));
+  while (headers && headers->EnumerateHeaderLines(&iter, &key, &value)) {
```

Comment:
The `headers` only needs to be tested for once, we can do it in the head of this function:

```c++
net::HttpResponseHeaders* headers = details.headers.get();
if (!headers)
  return;
```